File: Trees/Binary Tree Right Side View/iterative.py
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    // void helper(TreeNode* root, vector<int> & v, int level){
    //     if (root==NULL) return;
    //     else if (level>=v.size()) v.emplace_back(root->val);
    //     helper(root->right,v,level+1);
    //     helper(root->left,v,level+1);
    //     return;
    // }
    vector<int> rightSideView(TreeNode* root) {
        int level=0;
        vector<int> v;
        vector<int> s;
        vector<TreeNode *> stack;
        while (root || stack.size()>0){
            while(root){
                if (level>=v.size()) v.emplace_back(root->val);
                stack.emplace_back(root);
                s.emplace_back(level);
                root=root->right;
                level++;
            }    
            root = stack[stack.size()-1];
            level = s[s.size()-1];
            s.pop_back();
            stack.pop_back();
            if (root->left) {
                root=root->left;
                level+=1;
            }
            else root=NULL;
            
        }
        return v;
    }
};

File: Trees&Tries/Convert BST to Greater Tree/iterative.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* convertBST(TreeNode* root) {
    TreeNode *curr=root;
     int ret=0;
     vector<TreeNode*> stack;
     while(curr!=NULL || stack.size()>0){
        while (curr!=NULL){
            // cout << curr->val << endl;
            stack.emplace_back(curr);
            curr=curr->right;
        }
        curr=stack[stack.size()-1];
        // if (curr) cout << stack[stack.size()-1]->val << " " << ret<< endl;
        stack.pop_back();
        curr->val+=ret;
        ret=curr->val;
        curr=curr->left;
        
        
    }
        return root;
    }
};